src/main/java/io/logz/jmx2graphite/GraphiteClient.java

```diff
@@ -84,39 +90,60 @@ public static String sanitizeMetricName(String s, boolean keepDot) {
         }
         return sb.toString();
     }
-
-    /**
-     * send a list of metrics to graphite
-     * @param metrics
-     * @throws GraphiteWriteFailed
-     */
-
-    public void sendMetrics(List<MetricValue> metrics) throws GraphiteWriteFailed {
-        try {
-            if (!graphite.isConnected()) {
-                graphite.connect();
-            }
-        } catch (Exception e) {
-            throw new GraphiteWriteFailed("Failed connecting to Graphite. Error = "+e.getClass()+": "+e.getMessage(), e);
-        }
-
-        int failuresBefore = graphite.getFailures();
-        for (MetricValue mv : metrics) {
-            try {
-                graphite.send(metricsPrefix + mv.getName(), mv.getValue().toString(), mv.getTimestampSeconds());
-            } catch (Exception e) {
-                throw new GraphiteWriteFailed("Failed writing metric to graphite. Error = "+e.getMessage(), e);
-            }
-        }
-        try {
-            graphite.flush();
-        } catch (Exception e) {
-            throw new GraphiteWriteFailed("Failed writing metrics to graphite (flush). Error = "+e.getMessage(), e);
-        }
-        failuresAtLastWrite = graphite.getFailures() - failuresBefore;
+    
+    private boolean isBrokenPipeException(Throwable t) {
+    	if (t instanceof GraphiteWriteFailed) {
+    		Throwable cause = ExceptionUtils.getCause(t);
+    		String message = cause == null? t.getMessage() : cause.getMessage();
+    		return StringUtils.containsIgnoreCase(message, "broken pipe");
+    	}
+    	return false;
     }
 
-    public int getFailedAtLastWrite() {
+	/**
+	 * send a list of metrics to graphite
+	 * 
+	 * @param metrics
+	 * @throws GraphiteWriteFailed
+	 */
+	public void sendMetrics(List<MetricValue> metrics) throws GraphiteWriteFailed {
```

Comment:
Can you change indent here as well to 4 spaces as rest of class? It will make it easier to see what you changed in this method